### ISSUE TITLE
iOS in app purchase issue

### ISSUE THREAD

**Issue by ninexus-rajender** [AlexDisler/cordova-plugin-inapppurchase#199]
I've a IOS auto renewable subscriptions on ituneconnect for monthly bases, as in testflight mode everything is working fine, but in the live mode in app purchase is not working well, while we are purchasing subscription it's showing message as "**You're all set, your purchase was successful**" with in app purchase catch exception error popup behind itune payment success popup, it seems like response is not getting generated in time or is NULL which is why it's showing the exception, but on second time clicking the subscription it's saying "**you've already purchased this subscription**" and then it's returning the response properly, why is this working only second time on clicking the subscription as my app is on apple store and lot of users are paying for this app but they are facing this issue mentioned above, can you please provide some solution as any help on this will be appreciable.

**Comment by yashodeep9** [comment]
Hi,
I am trying to implement inapp purchase in iOS. I have created products and subscriptions in iTunes also created test user and released test flight mode. But when i try to buy the products i am getting error as undefined. Can you please guide me what i am missing here also do we need to accept the "Managing Agreements, Tax, and Banking Information" agreement even for the test flight mode ?

Thank you

**Comment by ninexus-rajender** [comment]
Yes you need to accept the agreements as mentioned by apple itunestore in order to start testing.

**Comment by yashodeep9** [comment]
Hi,
Thanks for your reply.
I accepted all the agreements and ready to submit the build to apple along with my inapp puchases.But the problem is i dont want to release this build update to existing users and wanted to test this inapp feature in testflight mode. Can you please guide me for this?

**Comment by shochberg** [comment]
If you set up a sandbox account with Apple and test that way, using a developer certificate, you do not need to upload your test app to Apple or get approval for your products or your app before you test.
